### PR TITLE
Solving memory leak issues due to opened cursor objects

### PR DESCRIPTION
**Pull request by guptasachin25**
@cmarcelk can you please review this code below. We were facing some issue due to memory leak in cordova plugins. I am submitting to apache first time. Let me know if i need to do something else to finally submit this code. 

Code changes explained: 
Cursor object was open for photoCursor in case of exceptions. 

Also cursor should be closed where it is created for consistency reasons. Added other two closed operations for same.

### CODE REVIEW

**Comment by guptasachin25** [comment]
Created this issue on JIRA. https://issues.apache.org/jira/browse/CB-9728

**Comment by guptasachin25** [comment]
@stevengill can you please review this change.

**Comment by shantanubhadoria** [comment]
+1 for this fix, I am facing the same issue here.

File: src/android/ContactAccessorSdk5.java [hunk 1]
```diff
@@ -283,6 +283,10 @@ public JSONArray search(JSONArray fields, JSONObject options) {
                 ContactsContract.Data.CONTACT_ID + " ASC");
          
         JSONArray contacts = populateContactArray(limit, populate, c);
+        
+        if (!c.isClosed()) {
+          c.close();
```
Comment (vladimir-kotikov) [inline src/android/ContactAccessorSdk5.java]: Please correct indentation to use 4 spaces per one level.

File: src/android/ContactAccessorSdk5.java [hunk 2]
```diff
@@ -313,7 +317,11 @@ public JSONObject getContactById(String id, JSONArray desiredFields) throws JSON
                 );
 
         JSONArray contacts = populateContactArray(1, populate, c);
-
+        
+        if (!c.isClosed()) {
+          c.close();
```
Comment (vladimir-kotikov) [inline src/android/ContactAccessorSdk5.java]: Same here

File: src/android/ContactAccessorSdk5.java [hunk 3]
```diff
@@ -945,9 +954,12 @@ private JSONObject photoQuery(Cursor cursor, String contactId) {
             }
         } catch (JSONException e) {
             Log.e(LOG_TAG, e.getMessage(), e);
-        } catch (SQLiteException e)
-        {
+        } catch (SQLiteException e) {
             Log.e(LOG_TAG, e.getMessage(), e);
+        } finally {
+          if(photoCursor != null && !photoCursor.isClosed()) {
```
Comment (vladimir-kotikov) [inline src/android/ContactAccessorSdk5.java]: Please correct indentation.

**Comment by vladimir-kotikov** [comment]
@guptasachin25, looks good.

Before merging this could you please address indentation issues and rebase this change on top of current master. Please also file an Apache CLA as described here: http://www.apache.org/licenses/#clas

**Comment by guptasachin25** [comment]
@vladimir-kotikov  Made changes and emailed signed A-CLA form. Let me know what next I can do to merge my changes

**Comment by vladimir-kotikov** [comment]
It looks like you hadn't pushed your changes, i don't see any updates.

**Pull request closed without merging**